### ISSUE TITLE
FunctionDriver<T> Component

### ISSUE THREAD

**Issue by Toxic-Cookie** [Frooxius/NeosPublic#1603]
## Is your feature request related to a problem? Please describe.
I always find myself needing to write LogiX just to drive a value field with a function like Sin. This is very inconvenient.

## Relevant issues
None.

## Describe the solution you'd like
A new component called something like FunctionDriver< T >. It should have three value fields where T is the numeric type. One is for the minimum value, one is for the maximum value, and one is for the speed. It should also have a reference field for the value field you will be driving just like other drivers. It should also have an enum with predefined functions such as Sin, Cos, Tan, etc. It would also make sense if this component obeyed its enabled field and only worked if it were enabled.

An example use would be as follows:
- Attach the component to an object. I'll put it on a basic cube.
- Specify T. I'll choose float3.
- Specify the minimum value. I'll choose (0, 0, 0).
- Specify the maximum value. I'll choose (1, 2, 1).
- Specify the speed. I'll choose (1 , 0.5, 1).
- Specify the the driven field. I'll choose the scale of the basic cube slot.
- Specify the Function enum. I'll choose Sin.

The result of the example would be a cube that would scale from (0, 0, 0) to (1, 2, 1) according to the Sin function at individual speeds of (1, 0.5, 1).

## Describe alternatives you've considered
Writing such a thing using LogiX works but, feels unnecessary and inconvenient. 

## Additional context
None.

**Comment by Frooxius** [comment]
This might be added, but it feels a bit too arbitrary and just duplicates functionality what's already available with LogiX. E.g. which functions would make the list? We'll have to choose a few common ones, but for any other setups you'd still have to use that.

It feels like setting up LogiX is the main source of inconvenience there. A better solution might be to create "canned" LogiX setups and have ability to quickly apply those to a field. That way you could prepare lots of different setups as needed, rather than being constrained to just a few.

Maybe one could be added for some common functions, but I'm just not sure which ones would make the list and which ones would not, it's not a solution that would scale too well.

**Comment by Toxic-Cookie** [comment]
Many components duplicate the functionality of LogiX. For example, the same argument could be made that a wobbler3D component is just a random float3 + a smooth lerp and thus unneeded.

This component request is pretty generalized and could be used in many situations. It would simplify the work of driving a value field with a common function.

The three most common functions would be Sin, Cos, and Tan. Others could be included as well and a reference to some can be found [here](https://www.mathsisfun.com/sets/functions-common.html).

**Comment by Frooxius** [comment]
Yeah that is true. If the operations are common enough then it'll be worth adding it, I'm just wondering if there's a way to generalize it more so it covers as many cases as possible or perhaps an even more general solution for the underlying problem.

**Comment by Frooxius** [comment]
Hmm, what functions should this ideally cover? I feel that it's best for any periodic functions. E.g. Sin/Cos, Triangle, Sawtooth, Square. Not fully sure about Tan, because at the extreme it goes all the way to infinity, so there's not really an "amplitude".

**Comment by Toxic-Cookie** [comment]
Just remembered that Blender [implements this in an excellent way.](https://docs.blender.org/manual/en/latest/animation/drivers/drivers_panel.html)

I'm not sure how exactly this could be translated into a component for Neos but it is well documented and worth looking at.
